cli/cmd/project/delete.go

```diff
@@ -25,24 +25,32 @@ func DeleteCmd(cfg *config.Config) *cobra.Command {
 			}
 			defer client.Close()
 
-			if !cmd.Flags().Changed("project") {
+			if len(args) > 0 {
+				name = args[0]
+			}
+
+			if !cmd.Flags().Changed("project") && len(args) == 0 && cfg.Interactive {
 				name, err = inferProjectName(cmd.Context(), client, cfg.Org, path)
 				if err != nil {
 					return err
 				}
 			}
```

Comment:
This is missing the select list for picking projects in interactive state. A new util for that could be something like:
```go
err = cmdutil.RequireSelectArgString(cmd, cfg, &name, args, 0, "project", "Select project", func () ([]string, error) {
  return cmdutil.ProjectNamesByOrg(ctx, client, cfg.Org)
  })
```

It would be like the previous suggested util, except it takes a func at the end that returns a list of `[]string` options to select from.

Then the code in this command (and many of the other `rill project` commands could be):

```go
if len(args) == 0 && !cmd.Flags().Changed("project") {
  name, err = inferProjectName(...)
  if err != nil {
    return err
  }
}

if name == "" {
  err = cmdutil.RequireSelectArgString(cmd, cfg, &name, args, 0, "project", "Select project", func () ([]string, error) {
    return cmdutil.ProjectNamesByOrg(ctx, client, cfg.Org)
  })
  if err != nil {
    return err
  }
}
```

Reply:
Yeah we can have some common util function select as well, but in above case `inferProjectName` has the SelectPrompt inside that function. So user will be prompted with project list when in interactive state.